src/main.rs

```diff
@@ -224,7 +222,7 @@ async fn handle_request(
     trace!("Response: {:?}", resp);
 
     #[cfg(feature = "expose-metrics")]
-    timing!(METRIC_KEY, start, end, "method"=>m.to_string(), "route"=>p, "status"=>resp.status().to_string());
+    histogram!(&METRIC_KEY[..], end - start, "method"=>m.to_string(), "route"=>p, "status"=>resp.status().to_string());
```

Comment:
`.as_ref()` ?

Reply:
This results in the error, any pointers?

```error[E0283]: type annotations needed
   --> src/main.rs:224:27
    |
224 |     histogram!(METRIC_KEY.as_ref(), end - start, "method"=>m.to_string(), "route"=>p, "status"=>resp.status().to_string());
    |                -----------^^^^^^--
    |                |          |
    |                |          cannot infer type for type parameter `T` declared on the trait `AsRef`
    |                this method call resolves to `&T`
    |
    = note: cannot satisfy `String: AsRef<_>`

error: aborting due to previous error
```

Reply:
Ah, guess the `histogram!` macro isn't picky enough about types. I tried `.as_str()` locally and that worked.

Reply:
Thanks, added to the PR